src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -2325,9 +2321,13 @@ bool IsVirtualNeededByTypeDueToPreservedScope (MethodDefinition method)
 			return false;
 		}
 
-		bool IsMethodNeededByTypeDueToPreservedScope (MethodDefinition method)
+		/// <summary>
+		/// Returns true if any of the base methods of <paramref name="method" /> is defined in an assembly that is not trimmed (i.e. action!=trim)
+		/// </summary>
+		/// <remarks>This is very similar to <see cref="IsVirtualNeededByTypeDueToPreservedScope(MethodDefinition)"/>, except this also checks if the base method is an interface.</remarks>
+		bool IsVirtualNeededByInstantiatedTypeDueToPreservedScope (MethodDefinition method)
```

Comment:
The new name is confusing - since this method explicitly handles interface methods as well (not just virtuals). I personally don't mind the previous method, it's not super descriptive, but at least it's not wrong.

Reply:
Actually with the removal of the call to this method from the delayed processing, this method should not handle static interface methods. Now it's only called when the type is instantiated - and in that case only virtual methods are affected. static interface methods must be marked regardless of instantiation.

Reply:
So the new name seems correct, but the method should not deal with statics at all.